### PR TITLE
Fine-tune log training times

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Logs detailed information for training/val/dataload/checkpointing times

Useful for checking optimal number of dataloader worker, batch size, etc.
The following is logged:
- Loggers (wandb etc):  Breakdown of train dataload/train/val dataload/val/checkpoint percentages
- Training step: Breakdown of dataload/forward/backward percentages
- Validation step: Breakdown of dataload/forward/metric percentages
- Debugging logger (train.log): Breakdown of all individual steps percentages

The goal is to give a high-level overview in the loggers and more detailed information in the individual steps and debugging logger.

Tensorboard output:
Screenshot 2026-02-06 at 11 43 27

train.log output:
```
[2026-02-06 11:55:27,760][INFO] Train Step 1/10 | Train Loss: 32.8633 | Time Spent [Dataload: 64.5%, Forward: 16.2%, Backward: 19.3%, Metrics: 0.0%]
[2026-02-06 11:55:30,879][INFO] Train Step 2/10 | Train Loss: 35.0486 | Time Spent [Dataload: 49.5%, Forward: 21.8%, Backward: 28.6%, Metrics: 0.0%]
[2026-02-06 11:55:50,765][INFO] Train Step 3/10 | Train Loss: 24.9726 | Time Spent [Dataload: 70.0%, Forward: 12.7%, Backward: 17.3%, Metrics: 0.0%]
[2026-02-06 11:55:53,866][INFO] Train Step 4/10 | Train Loss: 25.4516 | Time Spent [Dataload: 64.0%, Forward: 15.0%, Backward: 20.9%, Metrics: 0.0%]
[2026-02-06 11:56:15,433][INFO] Train Step 5/10 | Train Loss: 29.8776 | Time Spent [Dataload: 71.7%, Forward: 11.9%, Backward: 16.3%, Metrics: 0.0%]
[2026-02-06 11:56:15,435][INFO] Validating...
[2026-02-06 11:56:33,004][INFO] Val Step 1/2 | Time Spent [Dataload: 95.2%, Forward: 4.8%]
[2026-02-06 11:56:45,969][INFO] Val Step 2/2 | Val Loss: 3.0824 | Val mAP@0.5:0.95: 0.0000 | Val mAP@0.5: 0.0000 | Val mAP@0.75: 0.0000 | Val mAP (small): 0.0000 | Val mAP (medium): 0.0000 | Val mAP (large): 0.0000 | Time Spent [Dataload: 94.4%, Forward: 5.4%, Metrics: 0.2%]
[2026-02-06 11:56:45,971][INFO] The best validation metric val_metric/map=0.0000 was reached.
[2026-02-06 11:56:45,971][INFO] Saving the best checkpoint to '/Users/guarin/lightly-train/out/train_object_detection/2026-02-06_11-55-17/checkpoints/best.ckpt'
[2026-02-06 11:56:46,434][INFO] Exporting the best model to '/Users/guarin/lightly-train/out/train_object_detection/2026-02-06_11-55-17/exported_models/exported_best.pt'
[2026-02-06 11:56:46,593][DEBUG] Time Spent [Checkpoint Saving: 0.7%, Train Backward: 10.6%, Train Dataload: 46.5%, Train Forward: 7.8%, Train Metrics: 0.0%, Val Dataload: 32.5%, Val Forward: 1.9%, Val Metrics: 0.1%]
```

## How has it been tested?

* Unit tests
* Manually

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (internal change without effects for user)

### CODE REVIEW

File: src/lightly_train/_training_step_timer.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+import time
+
+
+class TrainingStepTimer:
+    """Timer for tracking time spent in different training steps."""
+
+    def __init__(self) -> None:
+        self._step_start_times: dict[str, float] = {}
+        self._step_last_durations: dict[str, float] = {}
```
Comment (yutong-xiang-97) [inline src/lightly_train/_training_step_timer.py]: Nit: `self._step_last_durations` only stores `duration`s and is not actively used elsewhere.

File: src/lightly_train/_commands/train_task_helpers.py [hunk 2]
```diff
@@ -683,10 +685,65 @@ def log_step(
...
+    times_perc = timer.percentage_for_prefix_group(
+        prefixes={
+            "train_dataload": ["train_dataload"],
+            "train": ["train"],
+            "val_dataload": ["val_dataload"],
+            "val": ["val"],
+            "checkpoint": ["checkpoint"],
+        }
+    )
+    log_dict["profiling/train_dataload_time_perc"] = times_perc["train_dataload"]
```
Comment (yutong-xiang-97) [inline src/lightly_train/_commands/train_task_helpers.py:740]: `add_timer_logs` is called in both train and validation steps where the two values can differ.

Comment (guarin) [reply src/lightly_train/_commands/train_task_helpers.py:740]: It measures times over the whole training/val loop so that is ok
